templates/common/vsphere/units/vsphere-hostname.service.yaml

```diff
@@ -7,7 +7,7 @@ contents: |
   After=vmtoolsd.service
 
   # Only run on first boot and only if the host does not have a name yet
-  ConditionPathExists=/etc/ignition-machine-config-encapsulated.json
```

Comment:
Needs to be a separate commit.

I don't think we need to check for `/etc/ignition-machine-config-encapsulated.json` at all - checking for `/etc/hostname` should be sufficient (nothing writes on FCOS afaik and users can redefine /etc/hostname)

Reply:
Agreed about the seperate commit.  Unfortunately, this was how I was able to test.  If we go forward I'll seperate them.

Having the check is very slightly cleaner as it wouldn't run on first boot, but getting rid of it is fine to me also...